src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -418,6 +431,16 @@ internal void CommitWriterChanges()
         internal void AddNodeTypeToSearchIndex(NodeSearchElement node, Document doc)
         {
             if (addedFields == null) return;
+            // During DynamoModel initialization, the index writer should still be valid here
+            // If the index writer is null and index not locked, it means the index writer has been disposed, e.g. DynamoModel finished initialization
+            // If the index writer is null and index locked, it means another Dynamo session is currently updating the search index
+            // Try to create a new index writer to amend the index
+            if (writer == null && !IndexWriter.IsLocked(this.indexDir))
```

Comment:
Shouldn't the check be IndexWriter.IsLocked(this.indexDir)?

Reply:
@reddyashish We want to create indexwirter when the index is not locked, if it is locked (by another process) the creation will fail anyway